service/controller/v13/resource/instance/create_scale_workers.go

```diff
@@ -0,0 +1,75 @@
+package instance
+
+import (
+	"context"
+
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/azure-operator/service/controller/v13/key"
+	"github.com/giantswarm/azure-operator/service/controller/v13/resource/instance/internal/state"
+	"github.com/giantswarm/microerror"
+)
+
+func (r *Resource) scaleUpWorkerVMSSTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	// Double the desired number of nodes in worker VMSS in order to
+	// provide 1:1 mapping between new up-to-date nodes when draining and
+	// terminating old nodes.
+	nodeCountFunc := func(_ int64) int64 {
+		return int64(key.WorkerCount(customObject) * 2)
+	}
+
+	err = r.scaleVMSS(ctx, customObject, key.WorkerVMSSName, nodeCountFunc)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	return CordonOldWorkers, nil
+}
+
+func (r *Resource) scaleDownWorkerVMSSTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	// Scale down to the desired number of nodes in worker VMSS.
+	nodeCountFunc := func(_ int64) int64 {
```

Comment:
I don't follow why we need that callback. Why can't we simply configure the delta to `scaleVMSS`?